src/hotspot/share/gc/shared/gcVMOperations.cpp

```diff
@@ -166,8 +166,15 @@ void VM_GC_HeapInspection::doit() {
       log_warning(gc)("GC locker is held; pre-dump GC was skipped");
     }
   }
+  WorkerThreads* workers = Universe::heap()->safepoint_workers();
+  if (workers != nullptr) {
+      // The GC provided a WorkerThreads to be used during a safepoint.
+      // Can't run with more threads than provided by the WorkerThreads.
+      const uint capped_parallel_thread_num = MIN2(_parallel_thread_num, workers->max_workers());
+      WithActiveWorkers with_active_workers(workers, capped_parallel_thread_num);
```

Comment:
`WithActiveWorkers` is an stack-obj, so it must be in the same scope as using workers.

Reply:
Oh, good catch. So can I do something like:

```
HeapInspection inspect;
if (workers != nullptr) {
    // The GC provided a WorkerThreads to be used during a safepoint.
    // Can't run with more threads than provided by the WorkerThreads.
    const uint capped_parallel_thread_num = MIN2(_parallel_thread_num, workers->max_workers());
    WithActiveWorkers with_active_workers(workers, capped_parallel_thread_num);
    inspect.heap_inspection(_out, capped_parallel_thread_num);
    inspect.heap_inspection(_out, workers);
} else {
    inspect.heap_inspection(_out, _parallel_thread_num);
    inspect.heap_inspection(_out, nullptr);
}
```

?

(I'm not familiar with the exact workings of StackObj, sorry)